### PR TITLE
[RFC] Add Makefile to run linter and formatter

### PR DESCRIPTION
**Pull request by keisuke-umezawa**
## Motivation

Provide one line command to run linter and formatter locally.

## Future tasks
- Run tests by make command
- Run bench by make command

### CODE REVIEW

File: .github/workflows/checks.yml [hunk 1]
```diff
@@ -22,13 +22,5 @@ jobs:
...
-    - name: flake8
-      run: flake8 .
-    - name: isort
-      run: isort . --check --diff
-    - name: mypy
-      run: mypy .
-    - name: blackdoc
-      run: blackdoc . --check --diff
+    - name: Lint
+      run: make lint
```
Comment (crcrpar) [inline .github/workflows/checks.yml:26]: The previous style might be better in that where the tool failed is clear.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -56,24 +56,32 @@ Coding style is checked with [flake8](http://flake8.pycqa.org) and [isort](https
...
 ```
 
 If you catch format errors, you can automatically fix them by auto-formatters.
 
 ```bash
-# Install auto-formatters.
-$ pip install .[checking]
-
-$ ./formats.sh 
+$ make fmt
```
Comment (crcrpar) [inline CONTRIBUTING.md:84]: I'm not sure `fmt` is common naming enough or not.

**Comment by keisuke-umezawa** [comment]
@c-bata If you have some opinion for Makefile, could you also review it?

**Comment by keisuke-umezawa** [comment]
The motivation of this PR is listing the commands used in development and making them as easy to execute as possible.

However, for the above motivation, we already had some solutions:

1. README.md
2. *.sh, *.py scripts

In the current situation, we do not have a strong demand to introduce Makefile. So, I will close this PR.

If we will have strong demand to introduce Makefile such as aggregating commands that are not need to be written as a script, we can reopen this PR.

**Pull request closed without merging**